zcash_client_sqlite/src/error.rs

```diff
@@ -41,6 +45,11 @@ pub enum SqliteClientError {
     /// A received memo cannot be interpreted as a UTF-8 string.
     InvalidMemo(zcash_primitives::memo::Error),
 
+    /// A requested rewind would violate invariants of the
+    /// storage layer. The payload returned with this error is
+    /// the safe rewind height.
+    RequestedRewindInvalid(BlockHeight),
```

Comment:
I suggest including both the safe rewind height and the requested height. (Non-blocking; this could be done separately, although it is a public API change.)